coalib/output/Logging.py

```diff
@@ -1,10 +1,26 @@
 from datetime import datetime
+from collections import defaultdict
 import json
 import io
 import logging
 import logging.config
 
 
+class CounterHandler(logging.Handler):
+    """
+    A logging handler which counts the number of calls
+    for each logging level.
+    """
+    level_called = None
+
+    def __init__(self, *args, **kwargs):
+        super(CounterHandler, self).__init__(*args, **kwargs)
+        CounterHandler.level_called = defaultdict(lambda: 0)
```

Comment:
0) Whoops. Old habits ;)

1) You're right. I initially didn't have it as a class member, but a regular instance member. But it turns out there is no convenient way to access the handlers of a logger, so I redesigned it. I'll implement your suggestions.